packages/jest-circus/src/types.ts

```diff
@@ -0,0 +1,261 @@
+      parentProcess: Process;
+    }
+  | {
+      // Action dispatched after everything is finished and we're about to wrap
+      // things up and return test results to the parent process (caller).
+      name: 'teardown';
+    };
+
+export type TestStatus = 'skip' | 'done' | 'todo';
+export type TestResult = {
+  duration: number | null | undefined;
+  errors: Array<FormattedError>;
+  invocations: number;
+  status: TestStatus;
+  location: {column: number; line: number} | null | undefined;
+  testPath: Array<TestName | BlockName>;
+};
+
+export type RunResult = {
+  unhandledErrors: Array<FormattedError>;
+  testResults: TestResults;
+};
+
+export type TestResults = Array<TestResult>;
+
+export type GlobalErrorHandlers = {
+  uncaughtException: Array<(exception: Exception) => void>;
+  unhandledRejection: Array<
+    (exception: Exception, promise: Promise<any>) => void
+  >;
+};
+
+export type State = {
+  currentDescribeBlock: DescribeBlock;
+  currentlyRunningTest: TestEntry | undefined | null; // including when hooks are being executed
+  expand?: boolean; // expand error messages
+  hasFocusedTests: boolean; // that are defined using test.only
+  // Store process error handlers. During the run we inject our own
+  // handlers (so we could fail tests on unhandled errors) and later restore
+  // the original ones.
+  originalGlobalErrorHandlers?: GlobalErrorHandlers;
+  parentProcess: Process | undefined | null; // process object from the outer scope
+  rootDescribeBlock: DescribeBlock;
+  testNamePattern: RegExp | undefined | null;
+  testTimeout: number;
+  unhandledErrors: Array<Exception>;
+  includeTestLocationInResult: boolean;
+};
+
+export type DescribeBlock = {
+  children: Array<DescribeBlock>;
+  hooks: Array<Hook>;
+  mode: BlockMode;
+  name: BlockName;
+  parent: DescribeBlock | undefined | null;
+  tests: Array<TestEntry>;
+};
+
+export type TestError =
+  | Exception
+  | Array<[Exception | undefined | null, Exception]>; // the error from the test, as well as a backup error for async
+
+export type TestEntry = {
+  asyncError: Exception; // Used if the test failure contains no usable stack trace
+  errors: TestError;
+  fn: TestFn | undefined | null;
+  invocations: number;
+  mode: TestMode;
+  name: TestName;
+  parent: DescribeBlock;
+  startedAt: number | undefined | null;
+  duration: number | undefined | null;
+  status: TestStatus | undefined | null; // whether the test has been skipped or run already
+  timeout: number | undefined | null;
+};
+
+export const STATE_SYM = (Symbol(
+  'JEST_STATE_SYMBOL',
+) as unknown) as 'STATE_SYM_SYMBOL';
+export const RETRY_TIMES = (Symbol.for(
+  'RETRY_TIMES',
+) as unknown) as 'RETRY_TIMES_SYMBOL';
+// To pass this value from Runtime object to state we need to use global[sym]
+export const TEST_TIMEOUT_SYMBOL = (Symbol.for(
+  'TEST_TIMEOUT_SYMBOL',
+) as unknown) as 'TEST_TIMEOUT_SYMBOL';
+
+// TODO Add expect types to @jest/types or leave it here
```

Comment:
yeah, move them to `@jest/types`

Reply:
I have taken these types from `expect` module. Should we move other `expect` types to `@jest/types` also?

Reply:
just the ones that are used. I'd rather move more types as needed than bloat `@jest/types`